controllers/discovery/discovery.go

```diff
@@ -57,6 +59,7 @@ type ResourceUri struct {
 	GVK          v1.GroupVersionKind
 	Namespace    string
 	Name         string
+	Port         string
```

Comment:
Is this only valid for Services and Containers? We might have to add this to a specific struct or have a custom query attribute here.

Reply:
+1, I will go back to the custom query attribute, which I had in the previous PR.